internal/pkg/service/stream/storage/node/coordinator/fileimport/operator.go

```diff
@@ -224,41 +224,22 @@ func (o *operator) importFile(ctx context.Context, file *fileData) {
 	defer cancel()
 
 	// Lock all file operations in the sink
-	lock := o.locks.NewMutex(fmt.Sprintf("operator.sink.file.%s", file.FileKey.SinkKey))
-	if err := lock.Lock(ctx); err != nil {
-		o.logger.Errorf(ctx, "cannot lock %q: %s", lock.Key(), err)
+	lock, unlocker := sinklock.LockSinkFileOperations(ctx, o.locks, o.logger, file.FileKey.SinkKey)
+	if unlocker == nil {
 		return
 	}
-	defer func() {
-		if err := lock.Unlock(ctx); err != nil {
-			o.logger.Warnf(ctx, "cannot unlock lock %q: %s", lock.Key(), err)
-		}
-	}()
-
-	var err error
-	if !file.IsEmpty {
-		// Import the file to specific provider
-		err = o.plugins.ImportFile(ctx, &file.File)
-		if err != nil {
-			err = errors.PrefixError(err, "error when waiting for file import")
-		}
-	}
-
-	// Update the entity, the ctx may be cancelled
-	dbCtx, dbCancel := context.WithTimeout(context.WithoutCancel(ctx), 30*time.Second)
-	defer dbCancel()
-
-	// If there is no error, switch file to the importing state
-	if err == nil {
-		err = o.storage.File().SwitchToImported(file.FileKey, o.clock.Now()).RequireLock(lock).Do(dbCtx).Err()
-		if err != nil {
-			err = errors.PrefixError(err, "cannot switch file to the imported state")
-		}
-	}
+	defer unlocker()
 
+	// Import file
+	err := o.doImportFile(ctx, lock, file)
 	// If there is an error, increment retry delay
 	if err != nil {
 		o.logger.Error(ctx, err.Error())
+
+		// Update the entity, the ctx may be cancelled
+		dbCtx, dbCancel := context.WithTimeout(context.WithoutCancel(ctx), 30*time.Second)
+		defer dbCancel()
```

Comment:
I don't like that we have now 2 db contexts, so if we adjust one dbContext and the other not, we will come into inconsistency.
Either add constant for `dbContextTimeout` or refactor it so we have just 1 dbCtx.

Reply:
The timeout can be configurable, ... but 2 contexts were created here by creating the `doImportFile` function, which improves the code and error handling.